CMakeLists.txt

```diff
@@ -43,26 +43,26 @@ set(MAN_DIR share/man/man1 CACHE PATH "Navit man path")
 add_definitions ("-DMAN_DIR=\"${MAN_DIR}\"")
 # LIB_DIR
 IF(UNIX AND NOT ANDROID AND NOT APPLE)
-    IF (NOT LIBDIR)
-        MESSAGE(STATUS "LIBDIR variable is not defined. It will be autodetected now.")
-        MESSAGE(STATUS "You can set it manually with -DLIBDIR=<value>")
-        # check 64 bit
-        IF (CMAKE_SIZEOF_VOID_P EQUAL 4)
-		SET(LIB_DIR lib/navit CACHE PATH "Navit 32bit bin path")
-            MESSAGE(STATUS "   LIB_DIR (highest subdirectory if LIBDIR) is set to '${LIB_DIR}'")
-        ELSE (CMAKE_SIZEOF_VOID_P EQUAL 4)
-		SET(LIB_DIR lib64/navit CACHE PATH "Navit 64bit bin path")
-            MESSAGE(STATUS "   LIB_DIR (highest subdirectory if LIBDIR) is set to '${LIB_DIR}'")
-        ENDIF (CMAKE_SIZEOF_VOID_P EQUAL 4)
-    ELSE (NOT LIBDIR)
-        GET_FILENAME_COMPONENT (LIB_DIR ${LIBDIR} NAME)
-        MESSAGE(STATUS "   LIB_DIR (highest subdirectory if LIBDIR) is set to '${LIB_DIR}'")
-    ENDIF (NOT LIBDIR)
+	IF (NOT LIBDIR)
+		MESSAGE(STATUS "LIBDIR variable is not defined. It will be autodetected now.")
+		MESSAGE(STATUS "You can set it manually with -DLIBDIR=<value>")
+		# check 64 bit
+		IF (CMAKE_SIZEOF_VOID_P EQUAL 4)
+			SET(LIB_DIR lib/navit CACHE PATH "Navit 32bit bin path")
+			MESSAGE(STATUS "   LIB_DIR (highest subdirectory if LIBDIR) is set to '${LIB_DIR}'")
+		ELSE (CMAKE_SIZEOF_VOID_P EQUAL 4)
+			SET(LIB_DIR lib64/navit CACHE PATH "Navit 64bit bin path")
+			MESSAGE(STATUS "   LIB_DIR (highest subdirectory if LIBDIR) is set to '${LIB_DIR}'")
+		ENDIF (CMAKE_SIZEOF_VOID_P EQUAL 4)
+	ELSE (NOT LIBDIR)
+		SET (LIB_DIR ${LIBDIR}/navit CACHE PATH "Navit lib path")
```

Comment:
LIBDIR is the standard GNU and cmake variable that tells the root of the directory where ALL the libs should be (in a subfolder per package usually). See the link I put in the original description.

LIB_DIR is where Navit itself should have its library, so it is supposed to be `$LIBDIR/navit`.

Reply:
The added navit directory is exactly my problem. On sailfish this would not be allowed. That's why I don't like adding a hard coded path here. If anything is manually provided we should just use that.
Whithout overriding LIB_DIR directly it's impossible to get rid of it. That's why I added the possibility to override LIB_DIR directly then. And used just that for Sailfish in the end (before I decided not to use modules at all)

With your change now, it's impossible again to put the .so files directly into /usr/lib because it always adds "navit" with no way to overome that. This way we would need another var to make it:
LIB_DIR = ${LIBDIR}/${LIB_ADD_DIR}
And then get rid of the hardcoded "navit" directories in the "LIBDIR not set" block above as well.
So:
Step 1: determine LIBDIR if not set
Step 2: make LIB_DIR equal to LIBDIR/LIB_ADD_DIR

Then for Sailfish I could use LIBDIR=share/ LIB_ADD_DIR="harbour-navit/lib/" to get the requested share/harbour-navit/lib. With -DCMAKE_INSTALL_PREFIX:PATH=/usr this makes the silly but required path /usr/share/harbour-navit/lib

Reply:
I think the fact that you added `LIB_DIR` is awesome, don't get me wrong and I think it's a perfectly correct use-case and we should keep it like this.
This change is more about the distros that use the GNU standards, to get the retro-compatibility with existing build systems.
The reason LIBDIR is overwritten in Debian is that we store the libs in `lib` even for 64bits versions.